src/circuits/joinsplit.tcc

```diff
@@ -1,430 +1,496 @@
-                **/
-                // We first allocate the root
-                merkle_root.reset(new libsnark::pb_variable<FieldT>);
-                merkle_root->allocate(pb, FMT(this->annotation_prefix, " merkle_root"));
-
-                // We allocate 2 field elements to pack each inputs nullifiers and each output commitments
-                for (size_t i = 0 ; i < NumInputs ; i++) {
-                    packed_inputs[i].allocate(
-                        pb, 1 + 1, FMT(this->annotation_prefix, " in_nullifier"));
-                }
-                for (size_t i = NumInputs ; i < NumInputs + NumOutputs ; i++) {
-                    packed_inputs[i].allocate(
-                        pb, 1 + 1, FMT(this->annotation_prefix, " out_commitment"));
-                }
-
-                // We allocate 1 field element to pack the value (v_pub_in)
-                packed_inputs[NumInputs + NumOutputs].allocate(pb, 1, FMT(this->annotation_prefix, " v_pub_in"));
-
-                // We allocate 1 field element to pack the value (v_pub_out)
-                packed_inputs[NumInputs + NumOutputs + 1].allocate(pb, 1, FMT(this->annotation_prefix," v_pub_out"));
+#include <boost/static_assert.hpp>
+#include <src/types/merkle_tree_field.hpp>
 
-                // The inputs are: [Root, NullifierS, CommitmentS, value_pub_in, value_pub_out]
-                // The root is represented on a single field element
-                // Each nullifier, and each commitment are in {0,1}^256 and thus take 2 field elements to be represented,
-                // while value_pub_in, and value_pub_out are in {0,1}^64, and thus take a single field element to be represented
-                const size_t nb_packed_inputs = (2 * (NumInputs + NumOutputs)) + 1 + 1;
-                const size_t nb_inputs = 1 + nb_packed_inputs;
-                pb.set_input_sizes(nb_inputs);
-                // ------------------------------------------------------------------------------ //
+using namespace libzeth;
 
-                // Initialize the digest_variables
-                for (size_t i = 0; i < NumInputs; i++) {
-                    input_nullifiers[i].reset(new libsnark::digest_variable<FieldT>(pb, HashT::get_digest_len(), FMT(this->annotation_prefix, " input_nullifiers[%zu]", i)));
-                }
-                for (size_t i = 0; i < NumOutputs; i++) {
-                    output_commitments[i].reset(new libsnark::digest_variable<FieldT>(pb, HashT::get_digest_len(), FMT(this->annotation_prefix, " output_commitments[%zu]", i)));
-                }
+template<
+    typename FieldT,
+    typename HashT,
+    typename HashTreeT,
+    size_t NumInputs,
+    size_t NumOutputs>
+class joinsplit_gadget : libsnark::gadget<FieldT>
+{
+private:
+    /*
+     * Multipacking gadgets for the inputs (nullifierS, commitmentS, val_pub_in,
+     *val_pub_out) (the root is a field element) `NumInputs + NumOutputs`
```

Comment:
clang-format does not seem to handle these comments very well, for some reason.  `//` is handled correctly.  Possibly a bug.
With `ReflowComments: "false"` I get:
```
    /*
         * Multipacking gadgets for the inputs (nullifierS, commitmentS, val_pub_in, val_pub_out) (the root is a field element)
         * `NumInputs + NumOutputs` because we pack the nullifiers (Inputs of JS = NumInputs), the commitments (Output of JS = NumOutputs)
         * AND the v_pub_out taken out of the mix (+1) AND the public value v_pub_in that is put into the mix (+1)
        **/
```

Reply:
Let's switch to `//` then if these are correctly handled by the tool

Reply:
Understood.  I make the changes and push here.  Thanks.